stdlib/public/TensorFlow/OpaqueHandles.swift

```diff
@@ -16,61 +16,117 @@
 
 import CTensorFlow
 
-/// `ResourceHandle` is the type used by ops and the `#tfop()` syntax to
-/// represent TensorFlow "resource" values.  It exists only to represent edges
-/// in TensorFlow graphs, and has no host-side representation (and thus no
-/// methods).
-public final class ResourceHandle {
-  private init() {
-    fatalError("ResourceHandle is a marker type that can never be created")
-  }
-}
-
-/// `VariantHandle` is the type used by ops and the `#tfop()` syntax to
-/// represent TensorFlow "variant" values.
-/// TODO: rename this source file as this is no longer "opaque".
-public final class VariantHandle {
-  public let cTensorHandle: CTensorHandle
-
+/// A class that implements common functionality for resource and variant handles.
+public class ResourceVariantHandleBase {
+	public let cTensorHandle: CTensorHandle
+	public let cIsResource: Bool
+	
   @usableFromInline
-  init(owning cTensorHandle: CTensorHandle) {
+  init(owning cTensorHandle: CTensorHandle, cIsResource: Bool) {
     self.cTensorHandle = cTensorHandle
+		self.cIsResource = cIsResource
   }
 
   deinit {
     debugLog("De-initializing TensorHandle.")
     TFE_DeleteTensorHandle(cTensorHandle)
-    debugLog("Returning from deinit of VariantHandle.")
+    debugLog("Returning from deinit of handle.")
   }
-}
 
-extension VariantHandle : TensorSendableReceivable {
-  @inlinable
-  static func receiveFromAccelerator(
-    _ computation: _TensorComputation,
-    _ tensorID: Int
-  ) -> VariantHandle {
-    debugLog("Receiving variant tensor of id \(tensorID).")
+  static public func getTFDataTypeName(_ isResource: Bool) -> String {
```

Comment:
1. It is unconventional to use the word `get` in Swift.

    See [Swift API Design Guidelines - Naming](https://swift.org/documentation/api-design-guidelines/#naming).

2. This is a local helper. It should not be public.

3. This refactoring is making the code harder to read. Essentially, this is a function that takes a boolean flag and returns a string without clear logic described in the name of the function or the type signature. It is better to keep `VariantHandle`/`ResourceHandle` conformances as is because these types will be cleared away through [SR-8523](https://bugs.swift.org/browse/SR-8523).